features/yield-calculator/YieldCalculator.tsx

```diff
@@ -0,0 +1,133 @@
+import { Box, TextField, Typography } from "@material-ui/core";
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import EmpState from "../../containers/EmpState";
+
+const Container = styled.div`
+  padding: 1rem;
+  border: 1px solid #434343;
+`;
+
+const FormInput = styled.div`
+  margin-top: 20px;
+  margin-bottom: 20px;
+`;
+
+const MS_PER_S = 1000;
+const S_PER_DAY = 60 * 60 * 24;
+const DAYS_PER_YEAR = 365;
+
+// TODO: When the yUSD market goes live on an exchange, we can replace this
+// `PLACEHOLDER_PRICE` with the real-time mid-market price for yUSD.
+const PLACEHOLDER_PRICE = "0.9875";
+
+// TODO: If the user has not selected an EMP (or has not even connected their web3 provider yet),
+// then the EMP's expirationTimestamp will clearly be unavailable. In order to support explorability,
+// this is a placeholder `daysToExpiry` that will be the default value for the textfield if no EMP is selected yet.
+const PLACEHOLDER_DAYS_TO_EXPIRY = "30";
+
+const YieldCalculator = () => {
+  const { empState } = EmpState.useContainer();
+  const [tokenPrice, setTokenPrice] = useState<string | null>(
+    PLACEHOLDER_PRICE
+  );
+  const [yieldAmount, setYieldAmount] = useState<number | null>(null);
+  const expirationTimestamp = empState.expirationTimestamp;
+
+  const calculateDaysToExpiry = () => {
+    if (expirationTimestamp) {
+      const currentTimestamp = Math.round(Date.now() / MS_PER_S);
+      const secondsToExpiry = expirationTimestamp.toNumber() - currentTimestamp;
+
+      return Math.round(secondsToExpiry / S_PER_DAY);
+    } else {
+      return null;
+    }
+  };
+
+  // @dev: We set this state var `daysToExpiry` after declaring `calculateDaysToExpiry()` because we first want to check if its value is
```

Comment:
nit: is it normal to put `@dev` in front of inline (as opposed to function) comments?

Reply:
No I just do it. I'll remove it